src/CoreBundle/Bridge/Chameleon/Module/Breadcrumb/BreadcrumbBackendModule.php

```diff
@@ -0,0 +1,323 @@
+use ChameleonSystem\CoreBundle\DataModel\BackendBreadcrumbItem;
+use ChameleonSystem\CoreBundle\DataModel\MenuCategoryAndItem;
+use ChameleonSystem\CoreBundle\Service\LanguageServiceInterface;
+use ChameleonSystem\CoreBundle\Util\UrlUtil;
+use IMapperCacheTriggerRestricted;
+use IMapperVisitorRestricted;
+use Symfony\Component\Debug\Exception\UndefinedMethodException;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class BreadcrumbBackendModule extends \MTPkgViewRendererAbstractModuleMapper
+{
+    /**
+     * @var MenuItemDataAccessInterface
+     */
+    private $menuItemDataAccess;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var UrlUtil
+     */
+    private $urlUtil;
+
+    /**
+     * @var LanguageServiceInterface
+     */
+    private $languageService;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    public function __construct(
+        MenuItemDataAccessInterface $menuItemDataAccess,
+        RequestStack $requestStack,
+        UrlUtil $urlUtil,
+        LanguageServiceInterface $languageService,
+        TranslatorInterface $translator
+    ) {
+        parent::__construct();
+
+        $this->menuItemDataAccess = $menuItemDataAccess;
+        $this->requestStack = $requestStack;
+        $this->urlUtil = $urlUtil;
+        $this->languageService = $languageService;
+        $this->translator = $translator;
+    }
+
+    // NOTE caching is difficult here because cache delete triggers would need to include virtually every table
+    //   as any entry's name might end up in the breadcrumb.
+
+    /**
+     * {@inheritDoc}
+     */
+    public function Accept(
+        IMapperVisitorRestricted $visitor,
+        $bCachingEnabled,
+        IMapperCacheTriggerRestricted $oCacheTriggerManager
+    ) {
+        $visitor->SetMappedValue('pathCmsRoot', PATH_CMS_CONTROLLER);
+
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (null === $request) {
+            return;
+        }
+
+        $currentUrl = $request->getRequestUri();
+        $urlParameters = $this->urlUtil->getUrlParametersAsArray($currentUrl);
+        $currentTableAndEntryId = $this->extractTableAndEntryId($urlParameters);
+        $tableConf = $this->getTableConf($currentTableAndEntryId[0] ?? null);
+
+        // NOTE see MTTableManager for potentially missing features (ie special cases for 'field' in url, one record redirect, bOnlyOneRecord, ...)
+
+        $tdbEntry = null;
+        if (null !== $tableConf && null !== $currentTableAndEntryId[1]) {
+            $tdbEntry = $tableConf->GetTableObjectInstance($currentTableAndEntryId[1]);
+        }
+
+        $parentTdb = null;
+        if (null !== $tableConf && null !== $tdbEntry) {
+            $parentTdb = $this->loadParent($tableConf, $tdbEntry, $urlParameters['sRestrictionField'] ?? null, $urlParameters['sRestriction'] ?? null);
+        }
```

Comment:
I think it would make sense to move the logic out of the module (=controller) and into a service. The Service could then return a data model that provides all relevant breadcrumb nodes.

Reply:
I think I'll wait with the rest of the review until we decided if the code should be moved. If it is moved, it becomes difficult to follow the review comments made earlier

Reply:
I was looking a bit into this and I don't see a clean/general/helpful interface:

The breadcrumb contents are somewhat complex depending for example on the pagedef being shown (i. e. sometimes a tdb is present for the last node, sometimes only a table and sometimes only something special like media manager).